exercise-book/src/nrf52-usb-listing-usb-devices.md

```diff
@@ -0,0 +1,18 @@
+# Listing USB Devices
+
+✅ To list all USB devices, run `cargo xtask usb-list` from the top-level checkout.
+
+```console
+$ cargo xtask usb-list
```

Comment:
1. This section asks users to have `cargo xtask` installed, but that isn't mentioned in the `Software Tools` previous page. 

Would you kindly add an appropriate installation entry for that as well?

2. `cargo xtask usb-lib` now emits a warning which will become a hard error:
```
❯ cargo xtask usb-lib
warning: user-defined alias `xtask` is shadowing an external subcommand found at: `C:\Users\mrg\scoop\persist\rustup-gnu\.cargo\bin\cargo-xtask.exe`
This was previously accepted but is being phased out; it will become a hard error in a future release.
For more information, see issue #10049 <https://github.com/rust-lang/cargo/issues/10049>.
```

I'm fine to file an issue and solve it later.

Reply:
`cargo xtask` is an alias for `cargo run --package ./xtask/Cargo.toml --` which is defined in the top level `.cargo/config.toml`. Happy to take suggestions on ways to make that clearer. I think Alex (used to work here) invented it: https://github.com/matklad/cargo-xtask

Reply:
Seeing you have a `.cargo/bin/cargo-xtask.exe` is troubling. Where did that come from?

Reply:
Ok, I 
1. uninstalled `cargo-xtask` from my global env and
2. reran this `cargo xtask usb-lib` now from root

and it now works as expected. I pattern matched greedily on `cargo $foo $bar` and thought I had to do `cargo install cargo-foo`, which pulled in that `.cargo/bin/cargo-xtask.exe`.

...my suggestion is now that "top-level checkout" could be replaced by "the crate root aka inside the `rust-exercises` folder".